### PR TITLE
adds class titles to each coach page

### PR DESCRIPTION
**Pull request by indirectlylit**
In preparation to potentially doing a full class-list switcher, as spec'd in the mockups.

This PR moves all class-related state out of pageState and to the top-level, so it's shared across pages.

image

### CODE REVIEW

**Comment by indirectlylit** [comment]
tests are failing - I still need to fix that. any quick advice would be appreciated

**Comment by rtibbles** [comment]
@indirectlylit looks like the ClassroomResource needs to be mocked in order for it to be callable in your generic class data fetching action? (which is then used across all the actions for the coach report page, I assume)

File: kolibri/plugins/coach/assets/src/state/actions/reports.js [hunk 1]
```diff
@@ -6,6 +6,7 @@ const { assessmentMetaDataState } = require('kolibri.coreVue.vuex.mappers');
 const CoreConstants = require('kolibri.coreVue.vuex.constants');
 const Constants = require('../../constants');
 const ReportConstants = require('../../reportConstants');
+const { setClassState } = require('./main');
```
Comment (DXCanas) [inline kolibri/plugins/coach/assets/src/state/actions/reports.js:9]: I'm really digging this syntax.

Comment (indirectlylit) [reply kolibri/plugins/coach/assets/src/state/actions/reports.js:9]: I think @jonboiser was the first to use that in our codebase

**Pull request merged**